sdmetrics/reports/utils.py

```diff
@@ -624,3 +625,79 @@ def print_results_for_level(out, results, level):
         out.write(f'\n{level}:\n')
         for result in results[level]:
             out.write(f'{level_marks[level]} {result}\n')
+
+
+def _validate_categorical_values(real_data, synthetic_data, metadata, table=None):
+    """Get categorical values found in synthetic data but not real data for all columns.
+
+    Args:
+        real_data (pd.DataFrame):
+            The real data.
+        synthetic_data (pd.DataFrame):
+            The synthetic data.
+        metadata (dict):
+            The metadata.
+        table (str, optional):
+            The name of the current table, if one exists
+    """
+    if table:
+        warning_format = 'Unexpected values ({values}) in column "{column}" '\
+                         f'and table "{table}"'
+    else:
+        warning_format = 'Unexpected values ({values}) in column "{column}"'
+
+    columns = get_columns_from_metadata(metadata)
+    for column, column_meta in columns.items():
+        column_type = get_type_from_column_meta(column_meta)
+        if column_type == 'categorical':
+            extra_categories = [
+                value for value in synthetic_data[column].unique()
+                if value not in real_data[column].unique()
+            ]
+            if extra_categories:
+                if len(extra_categories) > 5:
+                    value_list = '", "'.join(str(value) for value in extra_categories[:5])
+                    values = f'"{value_list}" + more'
+                else:
+                    value_list = '", "'.join(str(value) for value in extra_categories)
```

Comment:
minor: we could shorten this to
```python
value_list = '", "'.join(str(value) for value in extra_categories[:5])
values = f'"{value_list}" + more' if len(extra_categories) > 5 else f'"{value_list}"'
```
since taking a subset of a list with less elements than the max index doesn't crash, but the current way is maybe more explicit